From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 2 Aug 2026 23:51:02 +1000
Subject: [PATCH] arm64: dts: imx8mq-phanbell: add the 40-pin header I2S card

SAI1 goes to the 40-pin expansion header as plain I2S with no codec on
the far side. Describe it as a simple-audio-card against
linux,snd-soc-dummy, which is what the vendor kernel exposes as the
"40-pin Header" card.

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>
---
 .../boot/dts/freescale/imx8mq-phanbell.dts    | 39 +++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mq-phanbell.dts b/arch/arm64/boot/dts/freescale/imx8mq-phanbell.dts
index 9013e6e..470d0f9 100644
--- a/arch/arm64/boot/dts/freescale/imx8mq-phanbell.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mq-phanbell.dts
@@ -21,6 +21,25 @@
 		reg = <0x00000000 0x40000000 0 0x40000000>;
 	};
 
+	dummy_dai: dummy-dai {
+		compatible = "linux,snd-soc-dummy";
+		#sound-dai-cells = <0>;
+	};
+
+	sound-header {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "40-pin Header";
+		simple-audio-card,format = "i2s";
+
+		simple-audio-card,cpu {
+			sound-dai = <&sai1>;
+		};
+
+		simple-audio-card,codec {
+			sound-dai = <&dummy_dai>;
+		};
+	};
+
 	sound-analog {
 		compatible = "simple-audio-card";
 		simple-audio-card,name = "Coral Analog";
@@ -359,6 +378,16 @@
 	};
 };
 
+&sai1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sai1>;
+	assigned-clocks = <&clk IMX8MQ_CLK_SAI1>;
+	assigned-clock-parents = <&clk IMX8MQ_AUDIO_PLL1_OUT>;
+	assigned-clock-rates = <24576000>;
+	fsl,sai-mclk-direction-output;
+	status = "okay";
+};
+
 &sai2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_sai2>;
@@ -713,6 +742,16 @@
 		>;
 	};
 
+	pinctrl_sai1: sai1grp {
+		fsl,pins = <
+			MX8MQ_IOMUXC_SAI1_MCLK_SAI1_MCLK	0xd6
+			MX8MQ_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
+			MX8MQ_IOMUXC_SAI1_TXC_SAI1_TX_BCLK	0xd6
+			MX8MQ_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0	0xd6
+			MX8MQ_IOMUXC_SAI1_RXD0_SAI1_RX_DATA0	0xd6
+		>;
+	};
+
 	pinctrl_sai2: sai2grp {
 		fsl,pins = <
 			MX8MQ_IOMUXC_SAI2_MCLK_SAI2_MCLK	0xd6
